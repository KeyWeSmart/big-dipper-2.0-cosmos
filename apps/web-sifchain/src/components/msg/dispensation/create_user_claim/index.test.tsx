import renderer from 'react-test-renderer';
import CreateUserClaim from '@/components/msg/dispensation/create_user_claim';
import MsgCreateUserClaim from '@/models/msg/dispensation/msg_create_user_claim';
import MockTheme from '@/tests/mocks/MockTheme';

// ==================================
// mocks
// ==================================
jest.mock('@/components/name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Name" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/CreateUserClaim', () => {
  it('matches snapshot', () => {
    const message: MsgCreateUserClaim = {
      category: 'dispensation',
      type: 'MsgCreateDistribution',
      userClaimAddress: 'userClaimAddress',
      userClaimType: 'DISTRIBUTION_TYPE_AIRDROP',
      json: {},
    };
    const component = renderer.create(
      <MockTheme>
        <CreateUserClaim message={message} />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();

    expect(component.root.findByProps({ 'data-testid': 'Trans' }).props.i18nKey).toEqual(
      'message_contents:MsgCreateUserClaim'
    );
    expect(
      component.root.findByProps({ i18nKey: 'message_contents:MsgCreateUserClaim' }).props.values
        .userClaimType
    ).toEqual('DISTRIBUTION_TYPE_AIRDROP');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
