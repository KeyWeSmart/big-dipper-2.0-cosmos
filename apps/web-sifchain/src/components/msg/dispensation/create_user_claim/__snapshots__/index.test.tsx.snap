// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/CreateUserClaim matches snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<p
  className="MuiTypography-root MuiTypography-body1 emotion-0"
>
  <span
    data-testid="Trans"
    i18nKey="message_contents:MsgCreateUserClaim"
  >
    <div
      address="userClaimAddress"
      id="Name"
      name="userClaimAddress"
    />
    <b />
    {"userClaimType":"DISTRIBUTION_TYPE_AIRDROP"}
  </span>
</p>
`;
